packages/server/tests/sequencer.test.ts

```diff
@@ -0,0 +1,802 @@
+        expect(evts.length).toBe(count)
+        expect(evts.map((e) => e.seq)).toEqual(results[0].map((e) => e.seq))
+      }
+
+      lastSeen = results[0].at(-1)?.seq ?? lastSeen
+    })
+
+    it('respects abort signal during backfill', async () => {
+      // Create several events first to have something to backfill
+      for (let i = 0; i < 5; i++) {
+        await createDid()
+      }
+      await waitForSequencing()
+
+      const outbox = new Outbox(sequencer)
+      const abortController = new AbortController()
+
+      // Start from beginning to force backfill
+      const gen = outbox.events(0, abortController.signal)
+
+      const evts: SeqEvt[] = []
+
+      // Read a few events then abort
+      for await (const evt of gen) {
+        evts.push(evt)
+        if (evts.length >= 3) {
+          abortController.abort()
+          break
+        }
+      }
+
+      // Should have stopped after abort
+      expect(evts.length).toBeGreaterThanOrEqual(3)
+      expect(abortController.signal.aborted).toBe(true)
+    })
+
+    it('streams from the beginning when no cursor provided', async () => {
+      const outbox = new Outbox(sequencer)
+      const abortController = new AbortController()
+
+      // Start streaming without cursor (should skip straight to streaming mode)
+      const gen = outbox.events(undefined, abortController.signal)
+
+      // Create a new event - must happen after gen is started to be caught
+      const eventCreated = (async () => {
+        await wait(50) // Small delay to ensure gen is ready
+        await createDid()
+        await waitForSequencing()
+      })()
+
+      // Read the event with a timeout
+      const evts: SeqEvt[] = []
+      const timeout = setTimeout(() => abortController.abort(), 500)
+
+      try {
+        for await (const evt of gen) {
+          evts.push(evt)
+          if (evts.length >= 1) {
+            abortController.abort()
+            break
+          }
+        }
+      } catch {
+        // May throw on abort
+      }
+
+      clearTimeout(timeout)
+      await eventCreated
+
+      // Should have received the new event
+      expect(evts.length).toBeGreaterThanOrEqual(1)
+    })
+  })
+})
+
+describe('/export/stream endpoint', () => {
+  let server: TestServerInfo
+  let close: CloseFn
+  let db: Database
+  let sequencerLeader: SequencerLeader
+  let client: plc.Client
+  let wsUrl: string
+
+  beforeAll(async () => {
+    server = await runTestServer({
+      dbSchema: 'export_stream',
+    })
+
+    db = server.db
+    close = server.close
+    client = new plc.Client(server.url)
+    wsUrl = server.url.replace('http://', 'ws://') + '/export/stream'
+
+    // Start the sequencer leader
+    sequencerLeader = new SequencerLeader(db)
+    sequencerLeader.run().catch(() => {})
+  })
+
+  afterAll(async () => {
+    sequencerLeader?.destroy()
+    if (close) {
+      await close()
+    }
+  })
+
+  const createDid = async (): Promise<string> => {
```

Comment:
also nit, but maybe pull these into a `util` file or something to prevent duplications?